src/ui-components/input-field/input-field.spec.js

```diff
@@ -0,0 +1,315 @@
+          value: 123,
+        }
+      },
+      {
+        description: "when type is number and value is string",
+        givenData: {
+          id: undefined,
+          type: "number",
+          required: undefined,
+          value: "123",
+        },
+        expectedResults: {
+          id: undefined,
+          type: "number",
+          required: undefined,
+          value: "",
+        }
+      },
+      {
+        description: "when type is password and value is string",
+        givenData: {
+          id: undefined,
+          type: "password",
+          required: undefined,
+          value: "thisIsPassword",
+        },
+        expectedResults: {
+          id: undefined,
+          type: "password",
+          required: undefined,
+          value: "thisIsPassword",
+        }
+      },
+      {
+        description: "when id is given",
+        givenData: {
+          id: "testId123",
+          type: undefined,
+          required: undefined,
+          value: undefined,
+        },
+        expectedResults: {
+          id: "testId123",
+          type: "text",
+          required: undefined,
+          value: "",
+        }
+      },
+      {
+        description: "when required is true",
+        givenData: {
+          id: "",
+          type: undefined,
+          required: true,
+          value: undefined,
+        },
+        expectedResults: {
+          id: "",
+          type: "text",
+          required: true,
+          value: "",
+        }
+      },
+      {
+        description: "when required is false",
+        givenData: {
+          id: "",
+          type: undefined,
+          required: false,
+          value: undefined,
+        },
+        expectedResults: {
+          id: "",
+          type: "text",
+          required: false,
+          value: "",
+        }
+      }
+    ]
+
+    testCases.forEach(testCase => {
+      it(testCase.description, () => {
+        testInputFieldProps(testCase.givenData, testCase.expectedResults);
+      });
+    });
+
+  });
+
+    describe("validation messages", () => {
+      it("are rendered when provided only one", () => {
+        const testValidationMessages = ["validation message"]
+        const wrapper = shallow(<InputField validationMessages={testValidationMessages} />);
+
+        expect(wrapper.exists(`.${CSS_CLASSES.VALIDATION_MESSAGE}`)).toBeTruthy();
+        expect(wrapper.find(`.${CSS_CLASSES.VALIDATION_MESSAGE}`).text().trim()).toEqual(testValidationMessages[0]);
+      });
+      it("are rendered when provided when provided multiple", () => {
+        const testValidationMessages = ["validation message", "validation message 2", "validation message 3"];
+        const wrapper = shallow(<InputField validationMessages={testValidationMessages} />);
+
+        expect(wrapper.exists(`.${CSS_CLASSES.VALIDATION_MESSAGE}`)).toBeTruthy();
+        expect(wrapper.find(`.${CSS_CLASSES.VALIDATION_MESSAGE}`).length).toEqual(3);

```

Comment:
While both work here totally fine. `toEqual` is mostly used for object comparison. `toBe` should be used for values.

- [ ] Change of refute